jni/src/main/java/com/horizen/merkletreenative/MerklePath.java

```diff
@@ -0,0 +1,41 @@
+package com.horizen.merkletreenative;
+
+import com.horizen.librustsidechains.Library;
+import com.horizen.librustsidechains.FieldElement;
+
+public class MerklePath implements AutoCloseable {
+
+    private long merklePathPointer;
+
+    static {
+        Library.load();
+    }
+
+    private MerklePath(long merklePathPointer) {
+        if (merklePathPointer == 0)
+            throw new IllegalArgumentException("merklePathPointer must be not null.");
+        this.merklePathPointer = merklePathPointer;
+    }
+
+    private native boolean nativeVerify(int merkleTreeHeight, FieldElement leaf, FieldElement root);
```

Comment:
Why do we need the `merkleTreeHeight` here? Is it not enough just to apply the path elements to the leaf one by one and then compare it with the `root`?

Reply:
I want to check also that the path is of the correct length, otherwise I can just pass as a leaf a children of the root and the method will pass anyway.
As using the height as an explicit parameter, instead, is a result of keeping the height as a free parameter. Hardcoding the height in here means that if we want to modify the height of a tree, we should release a new version of the cryptolib